spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1461,11 +1448,12 @@ private Address getReplyToAddress(Message request) throws AmqpException {
 	private void addListener(Channel channel) {
 		if (channel instanceof PublisherCallbackChannel) {
 			PublisherCallbackChannel publisherCallbackChannel = (PublisherCallbackChannel) channel;
-			SortedMap<Long, PendingConfirm> pendingConfirms = publisherCallbackChannel.addListener(this);
 			Channel key = channel instanceof ChannelProxy ? ((ChannelProxy) channel).getTargetChannel() : channel;
-			if (this.pendingConfirms.putIfAbsent(key, pendingConfirms) == null
-					&& logger.isDebugEnabled()) {
-				logger.debug("Added pending confirms for " + channel + " to map, size now " + this.pendingConfirms.size());
+			if (this.publisherConfirmChannels.putIfAbsent(key, this) == null) {
```

Comment:
??? Do we need it?
We don't care here about the old value (value is always `this` BTW), so, simple `SetFromMap`:
```
public boolean add(E e) { return m.put(e, Boolean.TRUE) == null; }
```
is really enough for us.

Am I missing anything?

Although I agree: it is our internals and it will work the same way as right now. And it is always is based on the `this`, so no memory and GC impact as well.